packages/sitecore-jss-angular/src/components/editing-scripts.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, OnInit, Renderer2, Inject } from '@angular/core';
+import { getJssPagesClientData } from '@sitecore-jss/sitecore-jss/editing';
+import { JssStateService } from '../services/jss-state.service';
+import { DOCUMENT } from '@angular/common';
+import { EditMode, LayoutServicePageState } from '@sitecore-jss/sitecore-jss/layout';
+
+/**
+ * Component that renders editing scripts and client data for the current page in Sitecore Editor.
+ * Only renders scripts when Metadata mode is used.
+ */
+@Component({
+  selector: 'sc-editing-scripts',
+  template: '',
+})
+export class EditingScriptsComponent implements OnInit {
+  constructor(
+    private renderer: Renderer2,
+    private stateService: JssStateService,
+    @Inject(DOCUMENT) private document: Document
+  ) {}
+
+  ngOnInit(): void {
+    const state = this.stateService.stateValue;
+    const { pageState, editMode, clientData, clientScripts } = state.sitecore?.context || {};
+
+    // Don't render anything if not in editing/preview mode
+    if (pageState === LayoutServicePageState.Normal) return;
```

Comment:
Actually do we need to render client scripts/data in preview mode?

Reply:
Yes, it's still needed in "Preview" mode, since some scripts should be preloaded

Reply:
@yavorsk You were right, I double-checked. We don't need to care about Preview mode, but Preview mode uses default "editMode: Chromes" value, that allowed us to return an empty value. I added explicit checks to angular, react editing scripts components